heltour/tournament/models.py

```diff
@@ -1872,6 +1879,15 @@ def seed_rating_display(self, league=None):
             if league is None:
                 league = self.season.league
             return self.player.rating_for(league)
+    
+    def withdraw_from_team_season(self, round):
+        #we should only set players inactive here if they are not member of a team,
+        #i.e. if they are either alternates or we are in the pre-season before team creation.
+        if not TeamMember.objects.filter(player=self.player, team__season=round.season).exists():
+            self.is_active = False
+            self.save()
+        self._set_unavailable_for_season(skip_current=True)
+        add_system_comment(self, 'player withdrawn for %s'%round)
```

Comment:
Might want to make this “from round X” - also if this is in preseason what will the round be? Mostly for my ignorance here

Reply:
Withdrawals during pre-season are registered for round 1 (I have just double-checked that). The %s here resolves to a string "Season 31 - Round 1" for example. Maybe "for" and "from" are both not ideal but we could just make it "player withdrawn: %s"

Reply:
No this is perfectly fine with me then